cargo-quickinstall/tests/integration_tests.rs

```diff
@@ -0,0 +1,34 @@
+use cargo_quickinstall::*;
+
+#[test]
+fn do_dry_run_for_nonexistent_package() {
+    let crate_details = CrateDetails {
+        crate_name: "nonexisting_crate_12345".to_string(),
+        version: "99".to_string(),
+        target: "unknown".to_string(),
+    };
+
+    let result = do_dry_run(&crate_details);
+
+    let expected = r#""cargo" "install" "nonexisting_crate_12345" "--version" "99""#;
+    assert_eq!(expected, &result);
+}
+
+/// Tests dry run for Ripgrep.
+///
+/// A prebuilt package of Ripgrep version 13.0.0 for Linux x86_64 is known to
+/// exist.
+#[test]
+fn do_dry_run_for_ripgrep() {
+    let crate_details = CrateDetails {
+        crate_name: "ripgrep".to_string(),
+        version: "13.0.0".to_string(),
+        target: "x86_64-unknown-linux-gnu".to_string(),
+    };
+
+    std::env::set_var("CARGO_HOME", "/tmp/mock-cargo-root");
```

Comment:
It's fine for this PR, but be aware that `std::env::set_var()` is not thread safe. If another test sets it to something else, you might get unexpected results.

If it becomes a problem later, I think the standard solution is to create a shared init() function that is called at the top of each test, and sets up the env the same way for all tests (potentially protected behind a std::sync::Once::call_once(||{}))

Reply:
> but be aware that std::env::set_var() is not thread safe.

I have added another test covering ```install_crate()``` and this thread safety immediately became an issue.  I managed to avoid it for now by dropping the second call to ```std::env::set_var()```, but yes, we need a better fix.  The ```std::sync::Once```-based approach may not work as [mktemp](https://docs.rs/mktemp/latest/mktemp/) removes its tmp directory when it goes out of scope.  Perhaps we will just have to acquire a lock over the CARGO_HOME env variable.  Something for later.

On the upside, the test code looks Windows-proof although I haven't tested it there :)